cpp/src/parquet/encoder.cc

```diff
@@ -633,6 +669,7 @@ inline void DictEncoderImpl<DType>::Put(const T& v) {
   int32_t memo_index;
   PARQUET_THROW_NOT_OK(memo_table_.GetOrInsert(v, on_found, on_not_found, &memo_index));
   buffered_indices_.push_back(memo_index);
+  unencoded_data_bytes_ += sizeof(T);
```

Comment:
I'm a little suspect the idea here, since for non-byte-array type, these can just be caculated by the `buffered_indices_`.